src/environments.js

```diff
@@ -216,13 +216,20 @@ defineEnvironment("aligned", {
     let res = {
         type: "array",
         cols: [],
+        rowsAreLines: true,
     };
-    res = parseArray(context.parser, res);
+    res = parseArray(context.parser, res, "display");
+    // Count number of columns = maximum number of cells in each row.
+    // At the same time, prepend empty group {} at beginning of every second
+    // cell in each row (starting with second cell) so that operators become
+    // binary.
```

Comment:
This is so that operators at the start of cell will have the proper spacing around them.  What about operators at the end of a cell, or does that not happen in practice?